### PR TITLE
Some clean up of Lie Groups code

### PR DESCRIPTION
**Pull request by luisenp**

### CODE REVIEW

File: theseus/labs/lie_functional/se3.py [hunk 1]
```diff
@@ -118,16 +120,16 @@ def randn(
         device=device,
         requires_grad=requires_grad,
     )
-    return torch.cat((rotation, translation), dim=2)
+    ret = torch.cat((rotation, translation), dim=2)
+    ret.requires_grad_(requires_grad)
```
Comment (mhmukadam) [inline theseus/labs/lie_functional/se3.py:124]: You could similarly remove the two `requires_grad=requires_grad,` lines for `randn`.

File: theseus/labs/lie_functional/__init__.py [hunk 2]
```diff
@@ -2,3 +2,56 @@
...
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
+from typing import Callable, Optional, Sequence
+
+import torch
+
+import theseus.labs.lie_functional.se3 as _se3_impl
+import theseus.labs.lie_functional.so3 as _so3_impl
+from .constants import DeviceType
+from .lie_group import BinaryOperatorFactory, UnaryOperatorFactory
```
Comment (mhmukadam) [inline theseus/labs/lie_functional/__init__.py:12]: Any reason for explicit path first two and relative path for last two?

File: theseus/labs/lie_functional/__init__.py [hunk 3]
```diff
@@ -2,3 +2,56 @@
...
+
+_CheckFnType = Callable[[torch.Tensor], None]
+_RadnFnType = Callable[
```
Comment (mhmukadam) [inline theseus/labs/lie_functional/__init__.py:15]: typo? radn -> rand

File: theseus/labs/lie_functional/__init__.py [hunk 4]
```diff
@@ -2,3 +2,56 @@
...
+        if hasattr(module, "check_project_matrix"):
+            self.check_project_matrix: _CheckFnType = module.check_project_matrix
+        self.check_left_act_matrix: _CheckFnType = module.check_left_act_matrix
+        self.check_left_project_matrix: _CheckFnType = module.check_left_project_matrix
+        self.rand: _RadnFnType = module.rand
+        self.randn: _RadnFnType = module.randn
+
+
+se3_fns = LieGroupFns(_se3_impl)
+so3_fns = LieGroupFns(_so3_impl)
```
Comment (mhmukadam) [inline theseus/labs/lie_functional/__init__.py:57]: This is great!

On the front end the usage would be se3.exp() etc or something different. The _se3_impl going in is throwing me off I think.

File: theseus/labs/lie_functional/__init__.py [hunk 5]
```diff
@@ -2,3 +2,56 @@
...
+        bool,
+    ],
+    torch.Tensor,
+]
+
+
+# Namespace to facilitate type-checking downstream
+class LieGroupFns:
+    def __init__(self, module):
+        self.exp, self.jexp = UnaryOperatorFactory(module, "exp")
```
Comment (mhmukadam) [inline theseus/labs/lie_functional/__init__.py:30]: These functions get called internally after each op is defined in for example se3.py. Could we just instead call this class at the end of each file? Do we even need to call it if these are already being called below in this file?

**Review by mhmukadam — APPROVED** [review]
Looks great! Nice work brining consistency for Jacobian factory and consolidating function generation.

File: theseus/labs/lie_functional/se3.py (hunk 1 above)
Comment (luisenp) [reply theseus/labs/lie_functional/se3.py:124]: Ah sorry, I thought I did. Will fix.

File: theseus/labs/lie_functional/__init__.py (hunk 2 above)
Comment (luisenp) [reply theseus/labs/lie_functional/__init__.py:12]: I think this wasn't working for some reason.

File: theseus/labs/lie_functional/__init__.py (hunk 4 above)
Comment (luisenp) [reply theseus/labs/lie_functional/__init__.py:57]: Usage for functional stuff should be similar to what you have, but I'm still playing around with moving things in a separate PR. Let's ignore this for now.

File: theseus/labs/lie_functional/__init__.py (hunk 5 above)
Comment (luisenp) [reply theseus/labs/lie_functional/__init__.py:30]: Ah yes, sorry, my plan was to remove the original function calls, I forgot about this. Thanks for pointing it out!

File: theseus/labs/lie_functional/__init__.py (hunk 2 above)
Comment (mhmukadam) [reply theseus/labs/lie_functional/__init__.py:12]: Oh strange.

File: theseus/labs/lie_functional/__init__.py (hunk 3 above)
Comment (luisenp) [reply theseus/labs/lie_functional/__init__.py]: Fixed.

File: theseus/labs/lie_functional/__init__.py (hunk 2 above)
Comment (luisenp) [reply theseus/labs/lie_functional/__init__.py:12]: Fixed in another PR.

File: theseus/labs/lie_functional/se3.py (hunk 1 above)
Comment (luisenp) [reply theseus/labs/lie_functional/se3.py:122]: Fixed.

**Review by mhmukadam — APPROVED** [review]

**Review by fantaosha — APPROVED** [review]
LGTM.

**Pull request merged**